lib/otw_sanitize/embed_sanitizer.rb

```diff
@@ -122,6 +132,23 @@ def ensure_https
       end
     end
 
+    # If the embed is hosted on the Archive, it's Dewplayer and can be replaced
+    # with <audio> tag(s).
+    #
+    # Refer to https://archiveofourown.org/admin_posts/250.
+    def replace_dewplayer
```

Comment:
This will convert embed examples from [the old news post](https://archiveofourown.org/admin_posts/250). Open questions:

- It's possible to embed .swf files using `<object>` tags, but that seems to be broken on beta at the moment (???). Cursory debugging shows that this line doesn't work:
  https://github.com/otwcode/otwarchive/blob/0622729c9d4fca1ec7b611235bb5e318256ebd70/lib/otw_sanitize/embed_sanitizer.rb#L77
  because Sanitize currently doesn't return the `<object>` parent of the `<param>` tag, but instead a non-element.
- It's possible to embed the multi-variant of the player (https://archiveofourown.org/system/dewplayer/dewplayer-multi.swf) and specify multiple audio URLs separated by pipe characters (`|`). Should we try to handle this case? We didn't document this in the old news post.

Reply:
To answer my second question, yeah there are fics using the multi version, so let's handle that.

Reply:
Soooo, wait. For the .swf files embedded with `<object>`, were we theoretically allowing them to be embedded from anywhere on the web, or just certain safelisted hosts? If it was anywhere, yikes. If it was safelisted hosts, that makes sense -- a lot of places used `object` at one time, but judging from [the examples in this issue)(https://otwarchive.atlassian.net/browse/AO3-5192), only 8tracks still used it, which would explain why we had no idea it was broken (8tracks was mostly dead for a while and still isn't in great shape).

Reply:
Sorta... we allow embeds only from safelisted hosts, and for safelisted hosts, embeds can be any of `<object>`, `<embed>`, or `<iframe>`. There's no restriction that each safelisted host must go with a specific embed type. That we use `<embed>` for Archive-hosted players, or `<object>` for 8tracks, is just because we follow documented examples, but we can certainly [rewrite any `<embed>` as `<object>` or vice versa](https://web.archive.org/web/20201113034702/https://perishablepress.com/embed-flash-or-die-trying/).

Reply:
Ahh, good. As long as it's limited to safelisted hosts, that doesn't sound too terrible, even if it wasn't entirely intentional. 

Anyway, what exactly was the question about this part?

Reply:
Are we fine with ignoring potential `<object>` Dewplayer embeds, since no new `<object>` embeds can be created at the moment anyway?

Reply:
Yeah, I'd say if you varied that greatly from the code we gave you, it's up to you to fix it manually.

Reply:
Thanks, no more questions from me on this thread.